lib/shared/src/models/sync.ts

```diff
@@ -253,9 +264,11 @@ export function syncModels({
 
                                                 // Add the client models to the list of models.
                                                 data.primaryModels.push(
-                                                    ...maybeAdjustContextWindows(clientModels).map(
-                                                        createModelFromServerModel
-                                                    )
+                                                    ...maybeAdjustContextWindows(
+                                                        clientModels,
+                                                        isDotComUser,
+                                                        longContextWindowFlag
+                                                    ).map(createModelFromServerModel)
```

Comment:
```suggestion
                                                    ...maybeAdjustContextWindows(clientModels).map(
                                                        createModelFromServerModel
                                                    )
```
clientModels are configurable by the users on the client sides (IDE) and the limit should be configured by the users.